shortener/urls/static/js/index.js

```diff
@@ -0,0 +1,25 @@
+const main = document.querySelector("main");
+
+main.addEventListener('click', (e) => {
+    if (e.target.className === "copy-btn") {
+        copyURL(e);
+    }
+})
+
+function copyURL(e) {
+    const copyText = e.target.parentElement.textContent.trim();
+
+    navigator.permissions.query({ name: "clipboard-write" }).then(result => {
+        if (result.state == "granted" || result.state == "prompt") {
+            navigator.clipboard.writeText(copyText);
+        }
```

Comment:
Wow, your JS code for the copy url function is a lot neater than ours!  I think  we used the code you commented out.